src/map/clif.cpp

```diff
@@ -12355,9 +12355,7 @@ void clif_parse_StopAttack(int fd,struct map_session_data *sd)
 void clif_parse_PutItemToCart(int fd,struct map_session_data *sd)
 {
 	struct s_packet_db* info = &packet_db[RFIFOW(fd,0)];
-	if (pc_istrading(sd))
-		return;
-	if (!pc_iscarton(sd))
+	if (pc_istrading(sd) || !pc_iscarton(sd) || pc_cant_act2(sd))
```

Comment:
There is a bunch of checks inside pc_putitemtocart.
Maybe check if you can move those two checks into the other function?